client/dynamic_config.go

```diff
@@ -240,6 +230,33 @@ func createDynamicConfigDescriptorProto(knownModels []*ModelInfo, dependencyRegi
 	return
 }
 
+// DynamicConfigGroupFieldNaming computes for given known model the naming of configuration group proto field
+// containing the instances of given model inside the dynamic config describing the whole VPP-Agent configuration.
+// The json name of the field is the same as proto name of field.
+func DynamicConfigGroupFieldNaming(modelDetail *models.ModelInfo) string {
+	return fmt.Sprintf("%v%v", modulePrefix(models.ToSpec(modelDetail.Spec).ModelName()), configGroupSuffix)
+}
+
+// DynamicConfigKnownModelFieldNaming compute for given known model the (proto and json) naming of proto field
+// containing the instances of given model inside the dynamic config describing the whole VPP-Agent configuration.
+func DynamicConfigKnownModelFieldNaming(modelDetail *models.ModelInfo) (protoName, jsonName string) {
+	simpleProtoName := simpleProtoName(modelDetail.ProtoName)
+	protoName = simpleProtoName + repeatedFieldsSuffix
+	jsonName = simpleProtoName + repeatedFieldsSuffix
+	if !existsModelOptionFor("nameTemplate", modelDetail.Options) {
+		protoName = simpleProtoName
+		jsonName = simpleProtoName
+	}
+	configGroupName := DynamicConfigGroupFieldNaming(modelDetail)
+	compatibilityKey := fmt.Sprintf("%v.%v", configGroupName, simpleProtoName)
+	if newNames, found := backwardCompatibleNames[compatibilityKey]; found {
+		// using field names from hardcoded configurator.Config to achieve json/yaml backward compatibility
+		protoName = newNames.protoName
+		jsonName = newNames.jsonName
+	}
+	return protoName, jsonName
+}
```

Comment:
How about a little bit of refactoring? :)
```suggestion
func DynamicConfigKnownModelFieldNaming(modelDetail *models.ModelInfo) (protoName, jsonName string) {
	simpleProtoName := simpleProtoName(modelDetail.ProtoName)
	configGroupName := DynamicConfigGroupFieldNaming(modelDetail)
	compatibilityKey := fmt.Sprintf("%v.%v", configGroupName, simpleProtoName)

	if newNames, found := backwardCompatibleNames[compatibilityKey]; found {
		// using field names from hardcoded configurator.Config to achieve json/yaml backward compatibility
		protoName = newNames.protoName
		jsonName = newNames.jsonName
	} else if !existsModelOptionFor("nameTemplate", modelDetail.Options) {
		protoName = simpleProtoName
		jsonName = simpleProtoName
	} else {
		protoName = simpleProtoName + repeatedFieldsSuffix
		jsonName = simpleProtoName + repeatedFieldsSuffix
	}

	return protoName, jsonName
}
```